### PR TITLE
Remove deprecated pkg and update to flat config

### PR DESCRIPTION
**Pull request by MReyna12**

### CODE REVIEW

File: frontend/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      parser: tsParser,
+      parserOptions: {
+        ecmaVersion: 'latest'
+      },
+      globals: {
+        ...globals.browser,
+        ...globals.es2020,
+        ...globals.node,
+      }
+    }
```
Comment (rnathuji) [inline frontend/eslint.config.js:60]: I'm curious: What exactly does this `global` impact / what happens if we don't include it?

File: frontend/eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    rules: {
+      '@typescript-eslint/space-before-function-paren': [
+        'error',
+        {
+          anonymous: 'always',
+          named: 'never',
+          asyncArrow: 'always'
+        }
+      ],
+      'no-undef': 'off'
```
Comment (rnathuji) [inline frontend/eslint.config.js:29]: Does this `no-undef` rule impact anything? (removing it doesn't seem to create errors)

Comment (MReyna12) [reply frontend/eslint.config.js:31]: It does not - it is leftover from testing some features.

File: frontend/eslint.config.js (hunk 1 above)
Comment (MReyna12) [reply frontend/eslint.config.js:62]: I don't notice any different behavior when it is not included. This was included as carryover from the original implementation, which was a boilerplate eslint config file. Here is the purpose that globals serve per the eslint docs [here](https://eslint.org/docs/latest/use/configure/language-options#specifying-globals) and [here](https://eslint.org/docs/latest/use/configure/language-options#predefined-global-variables). Looks like if you have global variables and want particular ESLint rules to apply to said variables, then you need to implement `globals`.

I've opted to remove them, but we have at least captured how to implement them in the new schema if something arises later down the line.

**Review by rnathuji — APPROVED** [review]

**Pull request merged**